### PR TITLE
[BASE-60] Implement prometheus metric for kafka queue consumer

### PR DESCRIPTION
**Pull request by Xaelias**
## 🧪 Testing Steps / Validation
```
# HELP kafka_consumer_messages_processed_total Multiprocess metric
# TYPE kafka_consumer_messages_processed_total counter
kafka_consumer_messages_processed_total{kafka_address="127.0.0.1:9092",kafka_success="false",kafka_topic="test_topic"} 20.0
# HELP kafka_consumer_message_processing_time_seconds Multiprocess metric
# TYPE kafka_consumer_message_processing_time_seconds histogram
kafka_consumer_message_processing_time_seconds_sum{kafka_address="127.0.0.1:9092",kafka_success="false",kafka_topic="test_topic"} 0.007279703000002691
kafka_consumer_message_processing_time_seconds_bucket{kafka_address="127.0.0.1:9092",kafka_success="false",kafka_topic="test_topic",le="0.0001"} 0.0
kafka_consumer_message_processing_time_seconds_bucket{kafka_address="127.0.0.1:9092",kafka_success="false",kafka_topic="test_topic",le="0.00025"} 0.0
kafka_consumer_message_processing_time_seconds_bucket{kafka_address="127.0.0.1:9092",kafka_success="false",kafka_topic="test_topic",le="0.000625"} 19.0
kafka_consumer_message_processing_time_seconds_bucket{kafka_address="127.0.0.1:9092",kafka_success="false",kafka_topic="test_topic",le="0.0015625"} 20.0
kafka_consumer_message_processing_time_seconds_bucket{kafka_address="127.0.0.1:9092",kafka_success="false",kafka_topic="test_topic",le="0.00390625"} 20.0
kafka_consumer_message_processing_time_seconds_bucket{kafka_address="127.0.0.1:9092",kafka_success="false",kafka_topic="test_topic",le="0.009765625"} 20.0
kafka_consumer_message_processing_time_seconds_bucket{kafka_address="127.0.0.1:9092",kafka_success="false",kafka_topic="test_topic",le="0.0244140625"} 20.0
kafka_consumer_message_processing_time_seconds_bucket{kafka_address="127.0.0.1:9092",kafka_success="false",kafka_topic="test_topic",le="0.06103515625"} 20.0
kafka_consumer_message_processing_time_seconds_bucket{kafka_address="127.0.0.1:9092",kafka_success="false",kafka_topic="test_topic",le="0.152587890625"} 20.0
kafka_consumer_message_processing_time_seconds_bucket{kafka_address="127.0.0.1:9092",kafka_success="false",kafka_topic="test_topic",le="0.3814697265625"} 20.0
kafka_consumer_message_processing_time_seconds_bucket{kafka_address="127.0.0.1:9092",kafka_success="false",kafka_topic="test_topic",le="0.95367431640625"} 20.0
kafka_consumer_message_processing_time_seconds_bucket{kafka_address="127.0.0.1:9092",kafka_success="false",kafka_topic="test_topic",le="2.384185791015625"} 20.0
kafka_consumer_message_processing_time_seconds_bucket{kafka_address="127.0.0.1:9092",kafka_success="false",kafka_topic="test_topic",le="5.9604644775390625"} 20.0
kafka_consumer_message_processing_time_seconds_bucket{kafka_address="127.0.0.1:9092",kafka_success="false",kafka_topic="test_topic",le="14.901161193847656"} 20.0
kafka_consumer_message_processing_time_seconds_bucket{kafka_address="127.0.0.1:9092",kafka_success="false",kafka_topic="test_topic",le="+Inf"} 20.0
kafka_consumer_message_processing_time_seconds_count{kafka_address="127.0.0.1:9092",kafka_success="false",kafka_topic="test_topic"} 20.0
# HELP kafka_consumer_active_messages Multiprocess metric
# TYPE kafka_consumer_active_messages gauge
kafka_consumer_active_messages{kafka_address="127.0.0.1:9092",kafka_topic="test_topic",pid="58176"} 0.0
```

## ✅ Checks

- [x] CI tests (if present) are passing
- [x] Adheres to code style for repo
- [x] Contributor License Agreement (CLA) completed if not a Reddit employee

### CODE REVIEW

File: baseplate/frameworks/queue_consumer/kafka.py [hunk 1]
```diff
@@ -99,14 +129,26 @@ def __init__(
...
         on_success_fn: Optional[Handler] = None,
+        bootstrap_servers: str = "",
     ):
         self.baseplate = baseplate
         self.name = name
         self.handler_fn = handler_fn
         self.message_unpack_fn = message_unpack_fn
         self.on_success_fn = on_success_fn
+        # as far as I can tell, there is no way to extract the bootstrap servers from the
+        # consumer or the message itself. So wee need to have it passed from the parent
```
Comment (nsheaps) [inline baseplate/frameworks/queue_consumer/kafka.py]: 🔕  `wee` => `we`

File: baseplate/frameworks/queue_consumer/kafka.py [hunk 2]
```diff
@@ -99,14 +129,26 @@ def __init__(
...
+        self.bootstrap_servers = bootstrap_servers
 
     def handle(self, message: confluent_kafka.Message) -> None:
+        logger.error("handling stuff")
```
Comment (nsheaps) [inline baseplate/frameworks/queue_consumer/kafka.py]: please remove

File: baseplate/frameworks/queue_consumer/kafka.py [hunk 3]
```diff
@@ -99,14 +129,26 @@ def __init__(
...
+        prom_labels = KafkaConsumerPrometheusLabels(
+            kafka_address=self.bootstrap_servers,
+            kafka_topic=message.topic() if message.topic() is not None else "",
+        )
+        prom_success = "true"
```
Comment (nsheaps) [inline baseplate/frameworks/queue_consumer/kafka.py:148]: This implies that success is labeled differently if it were not prometheus. Assuming that's not accurate, can we drop the prefix here?

File: baseplate/frameworks/queue_consumer/kafka.py [hunk 4]
```diff
@@ -158,13 +202,21 @@ def handle(self, message: confluent_kafka.Message) -> None:
...
+            prom_success = "false"
             # let this exception crash the server so we'll stop processing messages
             # and won't commit offsets. when the server restarts it will get
             # this message again and try to process it.
             logger.exception(
                 "Unhandled error while trying to process a message, terminating the server"
             )
             raise
+        finally:
+            logger.error(prom_labels)
```
Comment (nsheaps) [inline baseplate/frameworks/queue_consumer/kafka.py]: Please remove in favor of an actual error message (assuming one was not logged) when handling the error, not in `finally`

File: baseplate/frameworks/queue_consumer/kafka.py [hunk 5]
```diff
@@ -158,13 +202,21 @@ def handle(self, message: confluent_kafka.Message) -> None:
 
                 context.metrics.gauge(f"{self.name}.{topic}.offset.{partition}").replace(offset)
         except Exception:
+            prom_success = "false"
```
Comment (nsheaps) [inline baseplate/frameworks/queue_consumer/kafka.py:204]: 🔕  I didn't make a big deal out of this elsewhere but it'd be great if this was an actual boolean and converted to a lower case string when actually consumed

File: baseplate/frameworks/queue_consumer/kafka.py [hunk 6]
```diff
@@ -402,13 +463,19 @@ def commit_offset(
             with context.span.make_child("kafka.commit"):
                 self.consumer.commit(message=message, asynchronous=False)
 
+        logger.error(self.consumer)
```
Comment (nsheaps) [inline baseplate/frameworks/queue_consumer/kafka.py]: please remove debugging statements

**Review by nsheaps — CHANGES_REQUESTED** [review]
Please remove the debugging code

File: baseplate/frameworks/queue_consumer/kafka.py (hunk 3 above)
Comment (Xaelias) [reply baseplate/frameworks/queue_consumer/kafka.py:148]: That only implies that this is how we track the prometheus success. Given that this value is only used by prometheus, I feel like this is accurate.

File: baseplate/frameworks/queue_consumer/kafka.py (hunk 4 above)
Comment (Xaelias) [reply baseplate/frameworks/queue_consumer/kafka.py]: I could do without the 👎  :-) These are left from debugging and were not intended to go through though.

File: baseplate/frameworks/queue_consumer/kafka.py (hunk 5 above)
Comment (Xaelias) [reply baseplate/frameworks/queue_consumer/kafka.py:204]: I don't have a strong opinion either way. They were handled as string in the first PRs that were merged so we just kept it consistent.

**Review by nsheaps — APPROVED** [review]

**Review by KTAtkinson — APPROVED** [review]

**Review by maeivysea — APPROVED** [review]

**Pull request merged**